### DISCUSSION TITLE
'Run validation with arguments' results in error

### DISCUSSION THREAD

**Discussion by nitishfy** [checkstyle/checkstyle#12394] [Q&A] [answered]
I've been trying to run the command onto checkstyle project via the command mentioned here: https://checkstyle.org/cmdline.html#Run_after_compilation

Every time I put this command onto my terminal, it leads to some error as shown below. Any suggestions on where I'm going wrong?

![checkstyleError](https://user-images.githubusercontent.com/86051118/201464913-93b0570b-6aef-4ab2-a20b-ae2f0ed31eb6.png)

**Comment by rnveach** [comment]
Are you using windows? Also please add which command your running.

**Reply by nitishfy** [reply]
Yes I'm using windows. I ran this command using IntelliJ terminal : `mvn exec:java -Dexec.mainClass="com.puppycrawl.tools.checkstyle.Main" \
    -Dexec.args="-c /sun_checks.xml src/main/java"`

**Reply by rnveach** [reply, marked as answer]
The commands in the wiki are more for linux. Windows needs extra string grouping for some reason as it sees the period as some type of delimiter.

The following command should work for Windows,
> mvn exec:java -D"exec.mainClass=com.puppycrawl.tools.checkstyle.Main" -D"exec.args=-c /sun_checks.xml src/main/java"

**Reply by romani** [reply]
Is it possible to make some JavaScript trick in browser to let user select example of it's OS ?
Our code base will contain both but user can click on something in HTML page to show example for windows.

**Reply by rnveach** [reply]
This is our page, so yes, it is possible. If we go down this, I think some form of tabs is best.

I would actually check first if Windows and Linux can use the exact same commands. If they can't be EXACTLY the same then this makes sense to do.

**Reply by romani** [reply]
They different on line wrap for sure.
Not all commands needed line wrap

**Reply by nitishfy** [reply]
I've run the command as mentioned in the docs on windows which didn't work well and resulted in error.

How about we raise a PR with modification of docs - adding context like:
`Depending on your OS, you can run the following commands:
Linux: mvn exec:java -Dexec.mainClass="com.puppycrawl.tools.checkstyle.Main" \ -Dexec.args="-c /sun_checks.xml src/main/java"
Windows: mvn exec:java -D"exec.mainClass=com.puppycrawl.tools.checkstyle.Main" -D"exec.args=-c /sun_checks.xml src/main/java"
`

WDYT?

**Reply by nrmancuso** [reply]
>WDYT?

This is too complicated; maven and java are OS agnostic. There should be some common command that can work for both. @NitishKumar06 have you tried command at https://github.com/checkstyle/checkstyle/discussions/12394#discussioncomment-4128615 ? This command works for me in both bash and zsh in latest Fedora.
